src/plugins/embeddable/public/lib/state_transfer/types.ts

```diff
@@ -17,48 +17,37 @@
  * under the License.
  */
 
-import { EmbeddableInput } from '..';
+import { Optional } from '@kbn/utility-types';
+import { EmbeddableInput, SavedObjectEmbeddableInput } from '..';
 
 /**
- * Represents a state package that contains the last active app id.
+ * A state package that contains information an editor will need to create or edit an embeddable then redirect back.
  * @public
  */
 export interface EmbeddableEditorState {
   originatingApp: string;
-  byValueMode?: boolean;
+  embeddableId?: string;
   valueInput?: EmbeddableInput;
+  byValueMode?: boolean;
```

Comment:
Do we actually need this `byValue` parameter? Or is the fact that `valueInput` exists enough to conclude it is by value?

Reply:
`valueInput` didn't exist in cases where the embeddable was newly created by value. I removed this parameter in favor of checking if `embeddableId` exists